### DISCUSSION TITLE
Suggest your module ideas here!

### DISCUSSION THREAD

**Discussion by ZodiusInfuser** [pimoroni/yukon#9] [Ideas]
Pimoroni Yukon launched with 10 modules, covering a range of high-power devices. We hope that these will be suitable for many people's projects, but there may be cases where additional functionality is needed. The advantage of a modular system is that more modules can be made over time to expand its capabilities.

### 07/12/2023 (Launch)

* [Audio Amp [Mono]](https://shop.pimoroni.com/products/audio-amp-module-for-yukon)
* [Bench Power [Step-down DAC]](https://shop.pimoroni.com/products/bench-power-module-for-yukon)
* [Big Motor + Encoder](https://shop.pimoroni.com/products/big-motor-encoder-module-for-yukon)
* [Dual Motor / Bipolar Stepper](https://shop.pimoroni.com/products/dual-motor-bipolar-stepper-module-for-yukon)
* [Dual Switched Power Output](https://shop.pimoroni.com/products/dual-switched-power-output-module-for-yukon)
* [LED Strip](https://shop.pimoroni.com/products/led-strip-module-for-yukon?variant=41187126640723)
* [Quad Servo [Regulated]](https://shop.pimoroni.com/products/quad-servo-module-for-yukon?variant=41187321479251)
* [Quad Servo [Direct]](https://shop.pimoroni.com/products/quad-servo-module-for-yukon?variant=41187321446483)
* [Serial Bus Servo](https://shop.pimoroni.com/products/serial-servo-module-for-yukon)
* [Proto](https://shop.pimoroni.com/products/proto-module-for-yukon)

### 17/12/2024

* [RM2 Wireless](https://shop.pimoroni.com/products/rm2-wireless-module-for-yukon)

## Suggesting New Modules

To suggest a new module, first consider whether your idea makes sense as a module. Here are a few questions to help with that:

* Can your idea not be achieved with the current offering of modules? Consider if any alternative functions or modes would work for you.
* Does your idea take advantage of the Yukon slot's four direct GPIO connections to the RP2040 and/or its high-power output? Module ideas that do not require high-power are fine, although I2C and SPI only ideas are more suitable for Breakout Garden.
* Is your idea not easily replicable with a Proto module and a few soldered on components?

If you answer favourably to all of these, please suggest your idea as a reply to this discussion.

Thanks

P.S As of 17/12/2024 we now offer a wireless solution in the form of the [RM2 Wireless Module](https://shop.pimoroni.com/products/rm2-wireless-module-for-yukon) 🎉

**Comment by KE6MTO** [comment]
Interested in a CAN module. Looks like there would two options for this, a SPI-based (https://learn.adafruit.com/adafruit-rp2040-can-bus-feather/overview) solution or a PIO-based (https://github.com/KevinOConnor/can2040) solution. I'm looking at a way to interface with an OpenMV H7, with my project already using both UARTs, CAN might be a good next choice. If you do, maybe have an option on the connector to have the CAN pins, ground and high voltage. That way a single cable to the next CAN device can deliver CAN and power.

**Comment by KE6MTO** [comment]
A revision to the serial servo module that as an additional header with the level shifted serial and 5v instead of the high voltage. The use case is for RC type of receivers that are 5v logic serial connections.

**Reply by ZodiusInfuser** [reply]
Which RC Receiver are you trying to use? I've been using an Flysky FS-A8S recently for a project and that runs completely fine off 3.3V.

**Reply by KE6MTO** [reply]
Good point, I will just have to try and see if the ones I have will work at 3.3v on the proto module. I have ExpressLRS modules (https://www.expresslrs.org/), in particular this lite receiver (https://betafpv.com/collections/expresslrs-series-accessories/products/elrs-lite-receiver) and this more traditional style one (https://betafpv.com/collections/expresslrs-series-accessories/products/elrs-micro-receiver).

**Comment by ZodiusInfuser** [comment]
From @spin on discord:

```
Can we expect a regulated version of the serial servo module, just like for pwm servos?
The yukon seems like a good platform for 4s lifepo4 batteries, but there aren't many serial servos that can handle 14V... having a 5A step down would be really nice
or at least, a way to feed regulated power into it
```

**Comment by KE6MTO** [comment]
What about a pre-built Bluetooth module?

**Reply by ZodiusInfuser** [reply]
As mentioned in the opening post,
> P.S We are aware that some form of wireless solution will be popular, so will be exploring this.

**Reply by KE6MTO** [reply]
Fair enough, when I saw wireless I was just thinking WIFI and not bluetooth as part of that. Sounds good!

**Reply by beetlegigg** [reply]
@ZodiusInfuser 
I wonder how far your exploration into this wifi module has progressed as I would be most interested in such a module.  Thus far I have alternatively linked a rpi picoW that communicates to Yukon by UART via a proto board hookup and also used a couple of ESP32's using ESP Now for remote access to the Yukon, again with the receiver ESP using a UART link to the Yukon.

The proto board worries me a bit as it brings out the high voltage pin holes in an 'accessible' way.   Maybe a simple UART board with tx, rx and ground might be a good and simple addition to the module board range.

A final though on this is thats its probably easy enough to create ones own proto board layout by designing a suitably sized PCB if you could let us know if there's a kicad symbol and name of the 18 pin connector.  Perhaps even the exact dimension of the module boards to save getting out my ruler :-).  Is there any info on this anywhere?
Thanks.

**Reply by ZodiusInfuser** [reply]
> @ZodiusInfuser I wonder how far your exploration into this wifi module has progressed as I would be most interested in such a module. Thus far I have alternatively linked a rpi picoW that communicates to Yukon by UART via a proto board hookup and also used a couple of ESP32's using ESP Now for remote access to the Yukon, again with the receiver ESP using a UART link to the Yukon.

Wireless exploration is still progressing I'm afraid. It's nice that you have been able to find solutions, but I acknowledge it is not ideal.

> The proto board worries me a bit as it brings out the high voltage pin holes in an 'accessible' way. Maybe a simple UART board with tx, rx and ground might be a good and simple addition to the module board range.

I would not do such a simple board like that, but the idea from @metanav is still on my mind.

> A final though on this is thats its probably easy enough to create ones own proto board layout by designing a suitably sized PCB if you could let us know if there's a kicad symbol and name of the 18 pin connector. Perhaps even the exact dimension of the module boards to save getting out my ruler :-). Is there any info on this anywhere? Thanks.

There's no KiCad symbol, but I'm more than happy to sharing module details for people to create their own boards when asked. It was just not a priority to share those details for launch, and it's not been until recently that anyone has asked me 😉. For instance, I was in DMs with a person on Twitter the other week who's been designing their own brushless motor driver for Yukon: https://x.com/RosmoRobot/status/1787422842532901284

The connector is one we had custom made for SMD assembly, but it is standard 2mm pitch, so could use regular through-hole pads (like that driver above). If someone really wanted to have their module use our connector for personal use, I'm pretty sure we could throw a few into a future order 🙂.

**Reply by beetlegigg** [reply]
@ZodiusInfuser 
Thanks for your response.  Ok since I put my Yukon into a bot I'm missing one on my desktop already so I will pop in an order tomorrow for another one with a couple of bits and pieces, thus your kind offer of throwing in a few of those connectors is most welcome 🙂.  I have a few simple ideas for homebew modules with some form of connector, may be some screw connectors or perhaps molex connectors in preference to DuPont ones. (my bot is an outdoor bot and quite liable to take a bit of a tumble and to be shaken about a bit).   The link to the brushless motor driver homebrew module is most interesting and I will study it later on.   

I will update you with the order number so you can intercept it before your order picking bots get the order shoved out the door to the Royal Mail for shipping.
  
The board you may do in relation to the @metanav idea would also be of interest BTW.

**Reply by ZodiusInfuser** [reply]
> @ZodiusInfuser Thanks for your response. Ok since I put my Yukon into a bot I'm missing one on my desktop already so I will pop in an order tomorrow for another one with a couple of bits and pieces, thus your kind offer of throwing in a few of those connectors is most welcome 🙂
> I will update you with the order number so you can intercept it before your order picking bots get the order shoved out the door to the Royal Mail for shipping.

Excellent! When you place your order, please immediately email `support @ pimoroni . com` to mention this (including your order number), and preferably place the order a few hours before our 2pm BST shipping cutoff. I will give them the heads up of this now too. You can also nudge me, but the best place to do that would be in the Yukon channel of the Pimoroni discord (it's a useful place to be in anyway).

**Reply by ZodiusInfuser** [reply]
@beetlegigg Actually, before you place your order. Let me check we've actually got stock of the connector. We should have, as I don't believe we've used 5000, but the system's saying there's 0.

**Reply by ZodiusInfuser** [reply]
@beetlegigg It turns out the system was right, and we really have gone through 5000 connectors 😬 . I have managed to pull 3 connectors off failed prototypes for you though, to throw in to your next order. If you let me know your contact details via DM (not public here!) or an email to support ahead of your order, I can get a note to include them added on your account.

**Reply by beetlegigg** [reply]
@ZodiusInfuser  - Thanks for that, much appreciated.  Email sent.

**Reply by ZodiusInfuser** [reply]
@KE6MTO We now offer the [RM2 Wireless Module](https://shop.pimoroni.com/products/rm2-wireless-module-for-yukon). I apologise for this taking a long time to come, but we felt waiting for an official wireless solution from Raspberry Pi would be better in the long run. Had we known it would be a whole year, then perhaps a different choice would have been made.

**Comment by metanav** [comment]
I suggest creating a screw terminal module that allows for the selection of either 5V or 3.3V power output. Level shifting would be a great addition. The 5V or 3.3V output current should be 1000mA or above.

Possible use cases:

- Encoder for dual motors ( 5V or 3.3V )
- Interfacing (over serial/I2C) and powering another microcontroller, e.g.,  5V/3.3V Arduino boards or similar
- 5V 2.4GHz RC receivers
- Slamtec RPLIDAR A1 or similar (>=5V input, may consume 1A current at peak)

I have created a prototype using the Proto module. However, it does not support 5V.

![proto_1](https://github.com/pimoroni/yukon/assets/7456984/79b0e8fc-a976-4530-b9e5-735fcacf5bd9)
![proto_2](https://github.com/pimoroni/yukon/assets/7456984/4aab7921-46fa-4cd3-a875-ffcc5ed03e3b)

**Reply by ZodiusInfuser** [reply]
This is an interesting idea. So it would have the four fast IO exposed via screw terminals (probably with Power and Ground for each two). The 5V power would need to be boosted up from the 3v3 rather than stepped down from the V+. I am not sure 1000mA output would be practical (not continuous anyway).

Do you know if the RPLIDAR A1 has a motor driver built in? I had a cheap 360 lidar and that needed an external motor driver set to about 80% duty cycle, or direct 3.3V. The IO needed 5V with level shifting though, otherwise the laser would not turn on.

**Reply by ZodiusInfuser** [reply]
Is a screw terminal better for this, or would the same style of 3x4 header as on the servo modules be suitable?

**Reply by metanav** [reply]
The RPLidar A1 motor is controlled by PWM and requires 5V power, but the logic voltage is 3.3V.
The screw terminals are more secure than pin headers for connecting on mobile robots or high vibrational platforms. When using an external device with female DuPont connectors, pin headers are a better choice if you don't want to make any changes.

**Reply by ZodiusInfuser** [reply]
Thanks for that. Pinout wise, it sounds to me like having both 3.3V and 5V exposed would be useful? I was just thinking it would be one or the other, perhaps being switchable. Would you want the logic level shifting to be set per pin, or is the whole block of four IO okay? I.e. they're all 3.3V or all 5V logic, no mix-n-match? I suspect there won't be enough board area for the latter.

**Reply by metanav** [reply]
I may have a biased perspective but I think the module should have
- 4 x 5V capable of high current 
- 4 x 3.3V
- 4 x GND
- 4 x FAST (logic level shift 3.3V to 5V or unshifted, see this Adafruit breakout https://www.adafruit.com/product/5649)

I know there isn't much space on a module. Why not create an I/O module with screw terminals and pin headers that span two slots on the Yukon? I believe this could help overcome the limitations.

**Comment by ZodiusInfuser** [comment]
Just to update anyone who's following this discussion, there's a new [RM2 Wireless Module](https://shop.pimoroni.com/products/rm2-wireless-module-for-yukon) now available 🎉

**Reply by metanav** [reply]
Awesome!
